crates/fj-viewer/src/gui.rs

```diff
@@ -14,22 +14,47 @@
 //!
 //! <https://github.com/gfx-rs/wgpu/issues/1492>
 
+use std::path::PathBuf;
+
+use crossbeam_channel::{Receiver, Sender};
+
 use fj_interop::status_report::StatusReport;
 use fj_math::{Aabb, Scalar};
 
-use crate::graphics::DrawConfig;
+use crate::{file_dialog::show_file_dialog, graphics::DrawConfig};
+
+/// Event that are passed between the event_loop and gui
+pub enum GuiEvent {
+    AskModel,
+    LoadModel(PathBuf),
+}
```

Comment:
Maybe I'm missing something, but it looks like `AskModel` is only sent to the GUI from the main loop, while `LoadModel` is only sent by the GUI to the main loop. Both `match` statements that handle `GuiEvent` only handle one variant, while ignoring the other.

It seems to me like `GuiEvent` shouldn't exist, and `AskModel`/`LoadModel` should be separate structs instead.